radix-engine/src/system/system_modules/costing/fee_table.rs

```diff
@@ -54,17 +81,321 @@ impl FeeTable {
         self.tx_blob_price_per_byte
     }
 
-    pub fn kernel_api_cost(&self, entry: CostingEntry) -> u32 {
-        match entry {
-            CostingEntry::Invoke { input_size } => FIXED_LOW_FEE + (10 * input_size) as u32,
-
-            CostingEntry::CreateNode { size } => FIXED_MEDIUM_FEE + (100 * size) as u32,
-            CostingEntry::DropNode { size } => FIXED_MEDIUM_FEE + (100 * size) as u32,
+    /// CPU instructions usage numbers obtained from test runs with 'resource_tracker` feature enabled
+    /// and transformed (classified and groupped) using convert.py script.
+    fn kernel_api_cost_from_cpu_usage(&self, entry: &CostingEntry) -> u32 {
+        ((match entry {
+            CostingEntry::AllocateNodeId {
+                entity_type,
+                virtual_node,
+            } => {
+                if !virtual_node {
+                    if entity_type.is_some() {
+                        match entity_type.unwrap() {
+                            EntityType::GlobalAccessController => 287,
+                            EntityType::GlobalAccount => 278,
+                            EntityType::GlobalFungibleResource => 287,
+                            EntityType::GlobalGenericComponent => 198,
+                            EntityType::GlobalIdentity => 276,
+                            EntityType::GlobalNonFungibleResource => 283,
+                            EntityType::GlobalPackage => 264,
+                            EntityType::GlobalValidator => 212,
+                            EntityType::InternalAccount => 278,
+                            EntityType::InternalFungibleVault => 284,
+                            EntityType::InternalGenericComponent => 283,
+                            EntityType::InternalKeyValueStore => 210,
+                            EntityType::InternalNonFungibleVault => 292,
+                            _ => 264, // average of above values
+                        }
+                    } else {
+                        264 // average of above values
+                    }
+                } else {
+                    // virtual_node
+                    106
```

Comment:
Interesting that cost for allocating node ID depends on entity type and virtualisation flag.

Reply:
Internally `allocate_node_id()` has additional code which creates new `NodeId` struct, which is missing from `allocate_virtual_node_id()` because it gets `NodeId` as a parameter, so this is why virtual node id takes less instructions.
Also there is variation of the instructions count over EntityType because of `ids.insert(node_id)` call which inserts new item to `BTreeSet` in `allocate_node_id()` function, most probably if memory allocation occurs to extend available capacity it takes longer. It looks that dependency of the instructions count from EntityType is rather random.
Because of above I think we should just take average of the `kernel_allocate_node_id` instruction calls instead of EntityType based match. @iamyulong WDYT?